### PR TITLE
Add check for the field being dereferenced and returned to prevent panic

### PR DESCRIPTION
**Pull request by nicktate**
### Description

 #### What does this pull request accomplish?
Adds check for `KubernetesVersion` key accessed in the `getMasterKubernetesVersion` function to prevent a segfault panic from dereferencing a `nil`

 #### What issue(s) does this fix?
* n/a

 #### Additional Considerations
The template in question causing the error is not valid and should be removed, but this adds additional sanity checking to prevent the crash and subsequently prints the ID of the invalid template.

 ### Testing
See additional tests handling edge case

### CODE REVIEW

File: resource/templates_test.go [hunk 1]
```diff
@@ -172,6 +197,9 @@ func TestGetMasterKubernetesVersion(t *testing.T) {
 	_, err := getMasterKubernetesVersion(types.Template{})
 	assert.Error(t, err, "get master version of empty template")
 
+	_, err = getMasterKubernetesVersion(tmplNilK8sVersion)
+	assert.Error(t, err, "get master version of empty template")
```
Comment (mattkelly) [inline resource/templates_test.go]: `of template with nil k8s version` or similar

File: resource/templates.go [hunk 2]
```diff
@@ -140,7 +140,7 @@ func getMasterKubernetesVersion(t types.Template) (string, error) {
 	}
 
 	for _, np := range t.Configuration.Variable {
-		if np.Default.KubernetesMode != nil &&
+		if np.Default.KubernetesMode != nil && np.Default.KubernetesVersion != nil &&
```
Comment (mattkelly) [inline resource/templates.go]: I actually don't think that we should do these nil checks here for required values. If the API is behaving correctly, this should never happen, especially at this layer (you can imagine autogenerated code checking for nil values and error reporting though).

...that said, this is currently useful and the best easy solution, so I'm happy to merge it.

Comment (nicktate) [reply resource/templates.go]: Yeah, I'm fine either way, I think if we don't want nil checks we should also remove the `KubernetesMode` check to be consistent. But right now we don't have that validation autogenerated as you said so this may be useful to prevent the panic.
